src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,239 @@
+    ov::Shape end_shape = tensor.get_shape();
+    end_shape[dim] = end_pos;
+    return ov::Tensor(tensor, start_shape, end_shape);
+}
+
+ov::npuw::LLMInferRequest::LLMInferRequest(const std::shared_ptr<ov::npuw::LLMCompiledModel>& compiled_model,
+                                           const ov::npuw::LLMCompiledModel::KVCacheDesc& kvcache_desc)
+    : ov::ISyncInferRequest(compiled_model),
+      m_kvcache_desc(kvcache_desc) {
+    LOG_DEBUG("ov::npuw::LLMInferRequest::LLMInferRequest()");
+    LOG_BLOCK();
+    m_kvcache_request = compiled_model->kvcache_compiled->create_infer_request();
+    m_prefill_request = compiled_model->prefill_compiled->create_infer_request();
+
+    for (auto input_port : m_prefill_request->get_compiled_model()->inputs()) {
+        m_prefill_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_prefill_request->get_compiled_model()->outputs()) {
+        m_prefill_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+
+    for (auto input_port : m_kvcache_request->get_compiled_model()->inputs()) {
+        m_kvcache_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_kvcache_request->get_compiled_model()->outputs()) {
+        m_kvcache_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::prepare_for_new_conversation() {
+    // FIXME: for input_ids it must be padding from tokenizer that not available from here
+    // Get it from NPUW options
+    LOG_DEBUG("ov::npuw::LLMInferRequest::prepare_for_new_conversation()");
+    LOG_BLOCK();
+
+    auto prefill_compiled = m_prefill_request->get_compiled_model();
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("attention_mask")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids")), 0u);
+    fill_tensor<int64_t>(m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask")), 0u);
+    m_kvcache_desc.num_stored_tokens = 0u;
+    LOG_DEBUG("Done");
+}
+
+void ov::npuw::LLMInferRequest::infer_prefill(ov::SoPtr<ov::ITensor> input_ids,
+                                              ov::SoPtr<ov::ITensor> attention_mask,
+                                              ov::SoPtr<ov::ITensor> position_ids) {
+    LOG_DEBUG("LLMInferRequest::infer_prefill()");
+    LOG_BLOCK();
+
+    prepare_for_new_conversation();
+
+    auto prefill_compiled = m_prefill_request->get_compiled_model();
+
+    auto padded_input_ids = m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids"));
+    const size_t offset = padded_input_ids->get_size() - input_ids->get_size();
+    std::copy_n(input_ids->data<int64_t>(),
+                input_ids->get_size(),
+                padded_input_ids->data<int64_t>() + offset);
```

Comment:
Before running `copy_n` I'd add an assert here that all three input tensors are actually `uint64_t`.

Reply:
`uint64_t`? not `int64_t`?

Reply:
if that's int64_t then it should be int64_t.

Reply:
Added in common method `infer`